packages/lesswrong/server/manualMigrations/2023-03-16-createUserActivities.ts

```diff
@@ -0,0 +1,27 @@
+import { getSqlClientOrThrow } from '../../lib/sql/sqlClient';
+import { registerMigration } from './migrationUtils';
+
+// This has to be done as a manual migration because of a bootstrapping issue (typeerrors block the automatic migration from running)
+registerMigration({
+  name: "createUserActivities",
+  dateWritten: "2023-03-16",
+  idempotent: true,
+  action: async () => {
+    const db = getSqlClientOrThrow();
+
+    const sql = `
+    CREATE TABLE IF NOT EXISTS "UserActivities" (
+      _id varchar(27) PRIMARY KEY,
+      "visitorId" varchar(27),
+      "type" text,
+      "startDate" timestamptz,
+      "endDate" timestamptz,
+      "activityArray" double precision[],
+      "schemaVersion" double precision DEFAULT 1,
+      "createdAt" timestamptz DEFAULT CURRENT_TIMESTAMP,
+      "legacyData" jsonb
```

Comment:
The thread was a DM, but the important thing was "I couldn’t get the migration to run because the types couldn’t be generated without the table existing, so I did it as a manual migration", which goes against my understanding of how it _should_ work, so there's a probably a bug. In particular, it should be possible to run `yarn generate` before `yarn migrate up`.

It's tricky to actally set this up to test now that the table exists and the migrations have already been run (and without knowing exactly what the original error was), so this is probably just something we should look out for/fix next time we need it.